catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/AccessibilityPreferencesActivityTest.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.scrollTo;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_ADDITIONAL_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_DRAGNDROP_DELAY;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_FONTFACE_VALUE_DYSLEXIC;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_HIGH_CONTRAST;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_HIGH_CONTRAST_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_ELEMENT_SPACING;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_LARGE_TEXT;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_ADDITIONAL_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_DRAGNDROP_DELAY;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_FONTFACE;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_HIGH_CONTRAST;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_HIGH_CONTRAST_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_ELEMENT_SPACING;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_ICONS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_LARGE_TEXT;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_MYPROFILE_STARTER_BRICKS;
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_STARTER_BRICKS;
+
+@RunWith(AndroidJUnit4.class)
+public class AccessibilityPreferencesActivityTest {
+
+	private List<String> settings = new ArrayList<>(Arrays.asList(ACCESS_LARGE_TEXT, ACCESS_HIGH_CONTRAST,
+			ACCESS_ADDITIONAL_ICONS, ACCESS_LARGE_ICONS, ACCESS_HIGH_CONTRAST_ICONS, ACCESS_LARGE_ELEMENT_SPACING,
+			ACCESS_STARTER_BRICKS, ACCESS_DRAGNDROP_DELAY));
+
+	@Rule
+	public BaseActivityInstrumentationRule<AccessibilityPreferencesActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(AccessibilityPreferencesActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
```

Comment:
first off i would consider it best practice to check what state the shared preferences are in, and leave them after the test in exactly this configuration again.
so read and save the values youre going to change, and put them back in teardown. see SettingsActivityTest for example

Reply:
Makes sense! Done.